Lesson15/selcan/script.js

```diff
@@ -0,0 +1,120 @@
+// 8. Use the `split` method to split a string into an array of substrings and print it to the console.
+const stringExercise8 =
+	'Algorithms are a fundamental part of computer science.';
+
+console.log('Exercise 8: ',stringExercise8.split(' '));
+const ex8Array = stringExercise8.split(' ');
+const updatedArray = ex8Array.map(word => word[0].toUpperCase()+ word.slice(1));
+console.log('result', updatedArray.join(' '));
+console.log('------------------------');
+// 9. Use the `join` method to join an array of strings into a single string and print it to the console.
+const arrayExercise9 = ['An', 'array', 'of', 'strings'];
+
+console.log('Exercise9.1: ', arrayExercise9.join(" "));
+console.log('Exercise9.2: ', arrayExercise9.join(""));
+console.log('Exercise9.2: ', arrayExercise9.join(" Excercise "));
+console.log('------------------------');
+// 10. Use the `replace` method to replace "can help" with "help" in the string stringExercise10 and print the result to the console.
+const stringExercise10 =
+	'Software design patterns can help simplify complex systems.';
+
+console.log('Exercise10: ', stringExercise10.replace("can help", "help"));
+console.log('------------------------');
+// 11. Use the `startsWith` method to check if strings below starts with a price, and print
+// the result to the console.
+const string1Exercise11 = '$9.99 - Sale! 50% off all items in store';
+const string2Exercise11 =
+	'Bundle deal: Save $50 when you buy two products together';
+const priceSymbol = '$';
+
+console.log('Exercise 11.1: ',string1Exercise11.startsWith(priceSymbol));
+console.log('Exercise 11.2: ',string2Exercise11.startsWith(priceSymbol));
+console.log('------------------------');
+// 12. Use the `endsWith` method to check if a string ends with a "!" (exclamation mark), and print the result to the console.
+const string1Exercise12 = '$9.99 - Sale! 50% off all items in store';
+const string2Exercise12 =
+	'Bundle deal: Save $50 when you buy two products together!';
+
+console.log('Exercise 12.1: ', string1Exercise12.endsWith("!"));
+console.log('Exercise 12.2: ', string2Exercise12.endsWith("!"));
+console.log('------------------------');
+// 13. Use the `includes` method to check if a string includes a "%" character and print
+// the result to the console.
+const string1Exercise13 = '$9.99 - Sale! 50% off all items in store';
+const string2Exercise13 =
+	'Bundle deal: Save $50 when you buy two products together!';
+const percentSymbol = '%';
+
+console.log('Exercise13: ',string1Exercise13.includes(percentSymbol));
+console.log('Exercise13: ',string2Exercise13.includes(percentSymbol));
+console.log('------------------------');
+// 14. Use the `substring` method to extract the word "Databases" from string stringExercise14
+// based on two indices and print it to the console.
+const stringExercise14 =
+	'Databases are an essential component of many applications.';
+
+console.log('Exercise 14: ',stringExercise14.substring(9));
```

Comment:
A more reliable approach to solve this problem would be something like this:
```
const startIndex = stringExercise14.indexOf('Databases');
const endIndex = startIndex + 'Databases'.length;
console.log('Ex. 14: ', stringExercise14.substring(startIndex, endIndex));
```

In this case, we rely on start and index being calculated instead of using a static value. This is something that we often need to do in real life since the data that we work with changes.